realm/realm-library/src/main/java/io/realm/internal/OsSharedRealm.java

```diff
@@ -198,6 +221,18 @@ private OsSharedRealm(long nativeSharedRealmPtr, OsRealmConfig osRealmConfig) {
         // This instance should never need notifications.
         this.realmNotifier = null;
         nativeSetAutoRefresh(nativePtr, false);
+
+        boolean foundParentSharedRealm = false;
+        for (OsSharedRealm sharedRealm : sharedRealmsUnderConstruction) {
```

Comment:
Not sure I fully understand this fix? This will only work if _all_ the temporary `SharedRealm` instances are being created from the Java side? I cannot tell from this PR if this is the case or not?

Reply:
The temp instances created in native code will be released immediately and guaranteed by RAII (not the stupid GC :) )

So we only need to worry about the instances created by java, see `runMigrationCallback` and `runInitializationCallback`. Also this constructor is designed for this purpose only and it is private method.